orion/pkg/domains/user.go

```diff
@@ -0,0 +1,69 @@
+package domains
+
+import (
+	"database/sql"
+	"errors"
+	"regexp"
+	"time"
+)
+
+var TABLE_USERS = "users"
+
+type User struct {
+	Id         uint
+	CreatedAt  time.Time    `db:"created_at"`
+	UpdatedAt  time.Time    `db:"updated_at"`
+	DeletedAt  sql.NullTime `db:"deleted_at"`
+	FirstName  string       `json:"firstName"`
+	LastName   string       `json:"lastName"`
+	MiddleName string       `json:"middleName"`
+	Email      string       `json:"email"`
+	Phone      string       `json:"phone"`
+	IsGuardian bool         `json:"isGuardian"`
+	GuardianId uint         `db:"program_id" json:"guardianId"`
```

Comment:
This can also be nullable, so change this to `sql.NullUint`... But taking a look at `golang/sql`, there is no `sql.NullUint`.

Do you feel comfortable implementing our own `sql.NullUint`?
You can follow the following example from the Go open source.
https://github.com/golang/go/blob/5476967b1a3d29fef4061999c00cadbec19ac0e3/src/database/sql/sql.go#L184-L208
I would say don't put this NullUint struct in the repo package. Otherwise, we will have a circular dependency. We might want to create our own sql_helper package instead. And have this user.go file import that sql_helper package.